pkg/skaffold/docker/parse_test.go

```diff
@@ -171,6 +176,10 @@ var ImageConfigs = map[string]*v1.ConfigFile{
 }
 
 func mockRetrieveImage(image string) (*v1.ConfigFile, error) {
+	if image == "base" {
+		panic("we should never retrieve images which are stage names")
```

Comment:
panic is considered a bad practice in testing as it stops the test execution - can we just return an `error`, or pass in `testing.T`?

Reply:
also assertion logic should not be embedded in the mock, this makes a test hard to understand.

Reply:
suggestion: reorganize the mock with usual "capture" semantics, that records the calls as a slice of strings and make the retrieved images an explicit assertion in the tests.